src/main/java/com/ibm/cloud/sdk/core/service/BaseService.java

```diff
@@ -397,14 +421,27 @@ public void setDefaultHeaders(final Map<String, String> headers) {
    * authentication errors after this token expires.
    *
    * @param iamOptions object containing values to be used for authenticating with IAM
+   * @deprecated Use setAuthenticator(AuthenticatorConfig) instead
    */
+  @Deprecated
   public void setIamCredentials(IamOptions iamOptions) {
-    this.tokenManager = new IamTokenManager(iamOptions);
+    setAuthenticator(iamOptions);
   }
 
-  private void clearIamCredentials() {
-    this.tokenManager = null;
+  /**
+   * Initializes a new Authenticator instance based on the input AuthenticatorConfig instance and sets it as
+   * the current authenticator on the BaseService instance.
+   * @param authConfig the AuthenticatorConfig instance containing the authentication configuration
+   */
+  protected void setAuthenticator(AuthenticatorConfig authConfig) {
```

Comment:
This should be `public` if we're directing users here from something like `setUsernameAndPassword()`

Reply:
> We're trying to add this feature this week for Java while Phil's out, so I'll go ahead and take over finishing this PR.

I'm continuing to work on it so should have something by end of week.

Reply:
> This should be `public` if we're directing users here from something like `setUsernameAndPassword()`

Actually my intent was for setAuthenticator() to be called within the generated service class ctor.